Corgibytes.Freshli.Cli/Functionality/Git/GitArchive.cs

```diff
@@ -22,38 +27,67 @@ public class GitArchive
         // If it exists, make sure to empty it so we are certain we start with a clean slate.
         var gitSourceTarget = new DirectoryInfo(Path.Combine(cacheDirectory, "histories", gitSource.Id,
             gitCommitIdentifier.ToString()));
-        if (Directory.Exists(gitSourceTarget.FullName))
+
+        Task<string>? createTask = null;
+        lock (s_gitIdsAndSourceTargetsLock)
         {
-            Directory.Delete(gitSourceTarget.FullName, true);
+            if (s_gitIdsAndSourceTargets.ContainsKey(gitSource.Id))
+            {
+                createTask = s_gitIdsAndSourceTargets[gitSource.Id];
+            }
         }
 
-        // Create the directory where we want to place the archive
-        gitSourceTarget.Create();
-        var archivePath = Path.Combine(gitSourceTarget.FullName, "archive.zip");
+        if (createTask != null)
+        {
+            return createTask.Result;
+        }
 
-        var archiveProcess = new Process
+        lock (s_gitIdsAndSourceTargetsLock)
         {
-            StartInfo = new ProcessStartInfo
+            s_gitIdsAndSourceTargets.Add(gitSource.Id, new Task<string>(() =>
```

Comment:
I think a great way to shorten this method and address the "too long" issue raised by Code Climate is to extract the body of this task into a private method. What are your thoughts?

Reply:
Sounds good to me!